ttnn/cpp/ttnn/operations/matmul/device/matmul_op.cpp

```diff
@@ -973,7 +973,7 @@ void Matmul::validate(
         TT_FATAL(
             bias_shape[-2] == TILE_HEIGHT, "Unsupported bias shape: second last dimension not equal to tile height");
         TT_FATAL(
-            bias_shape[-1] == b_shape[-1],
+            bias_shape.without_padding()[-1] == b_shape[-1],
```

Comment:
On line 970, `const auto& bias_shape = bias.get_legacy_shape();` should be changed to `const auto& bias_shape = bias.get_shape();` and then this can stay as is.

Reply:
That would also require that the unpadded `bias_shape[-2] == TILE_HEIGHT`
@TT-BrianLiu could you please confirm that this is what we'd want?

Reply:
updated.

Reply:
If tests don't break, i think unpadded shape is fine for bias? At least for matmul, I don't think the biases need any padding AFAIK

Reply:
looks like i need to use the `without_padding` version as the other asserts need shape with padding.

Reply:
Where does it need shape with padding?

Reply:
line 976 -- it fails with shape from `get_shape`

Reply:
I meant where in the workflows (post commit, models, nightly, etc.)/tests.

Reply:
It was in post commit tests.

Reply:
https://github.com/tenstorrent/tt-metal/actions/runs/10496314234/job/29077332075

Reply:
```
XFAIL tests/ttnn/unit_tests/operations/test_max_pool2d.py::test_pool_core_nondivis[batch_size=1-input_channels=32-input_height=23-input_width=23-filter_height=3-filter_width=3-stride_h=1-stride_w=1-pad_h=1-pad_w=1-config_override={'num_cores_nhw': 10, 'snap_to_tile': True}-xfail=True-device_params={'l1_small_size': 24576}] - reason: 
FAILED tests/ttnn/unit_tests/operations/test_new_conv2d.py::test_resnet50_conv_gs[math_fidelity=MathFidelity.LoFi-activations_dtype=DataType.BFLOAT16-weights_dtype=DataType.BFLOAT16-batch_size=8-output_channels=64-input_channels=64-input_height=56-input_width=56-filter_height=1-filter_width=1-stride_h=1-stride_w=1-pad_h=0-pad_w=0-use_1d_systolic_array=True-config_override=None-device_params={'l1_small_size': 16384}] - RuntimeError: TT_FATAL @ ../ttnn/cpp/ttnn/operations/matmul/device/matmul_op.cpp:974: bias_shape[-2] == TILE_HEIGHT
info:
Unsupported bias shape: second last dimension not equal to tile height
```

Reply:
```
tests/ttnn/unit_tests/operations/test_new_conv2d.py::test_sd_conv[enable_auto_formatting=True-math_fidelity=MathFidelity.LoFi-activations_dtype=DataType.BFLOAT8_B-weights_dtype=DataType.BFLOAT8_B-batch_size=2-output_channels=320-input_channels=960-input_height=64-input_width=64-filter_height=1-filter_width=1-stride_h=1-stride_w=1-pad_h=0-pad_w=0-use_1d_systolic_array=True-config_override=None-device_params={'l1_small_size': 16384}]                   Metal | INFO     | Initializing device 0. Program cache is NOT enabled
                  Metal | INFO     | AI CLK for device 0 is:   1000 MHz
                  Metal | INFO     | Enabling program cache on device 0
                 Always | FATAL    | Unsupported bias shape: second last dimension not equal to tile height
FAILED                  Metal | INFO     | Disabling and clearing program cache on device 0
                  Metal | INFO     | Closing device 0
                  Metal | INFO     | Disabling and clearing program cache on device 0


=================================== FAILURES ===================================
_ test_sd_conv[enable_auto_formatting=True-math_fidelity=MathFidelity.LoFi-activations_dtype=DataType.BFLOAT8_B-weights_dtype=DataType.BFLOAT8_B-batch_size=2-output_channels=320-input_channels=960-input_height=64-input_width=64-filter_height=1-filter_width=1-stride_h=1-stride_w=1-pad_h=0-pad_w=0-use_1d_systolic_array=True-config_override=None-device_params={'l1_small_size': 16384}] _
```